otx/core/data/action_dataset_adapter.py

```diff
@@ -0,0 +1,199 @@
+        then the overall label should include all label informations as [0, 1, 2, 4].
+
+        [Reindexing the each label index of multiple video datasets]
+        In this case, if the label for 'video1/frame_000.jpg' is 2, then the index of label is set to 2.
+        For the case of video2, if the label for 'video2/frame_000.jpg' is 4, then the index of label is set to 2.
+        However, Since overall labels are [0, 1, 2, 4], 'video2/frame_000.jpg' should has the label index as 3.
+
+        """
+        outputs = {
+            "category_items": [],
+            "label_groups": [],
+            "label_entities": [],
+        }  # type: dict
+
+        category_list = []  # to check the duplicate case
+        for cvat_data in datumaro_dataset[Subset.TRAINING]:
+            # Making overall categories
+            categories = cvat_data.categories().get(DatumaroAnnotationType.label, None)
+
+            if categories not in category_list:
+                outputs["category_items"].extend(categories.items)
+                outputs["label_groups"].extend(categories.label_groups)
+
+            category_list.append(categories)
+
+            # Reindexing the each label index of multiple video datasets
+            for cvat_data_item in cvat_data:
+                for ann in cvat_data_item.annotations:
+                    ann_name = categories.items[ann.label].name
+                    ann.label = [
+                        i for i, category_item in enumerate(outputs["category_items"]) if category_item.name == ann_name
+                    ][0]
+
+        # Generate label_entity list according to overall categories
+        outputs["label_entities"] = [
+            LabelEntity(name=class_name.name, domain=self.domain, is_empty=False, id=ID(i))
+            for i, class_name in enumerate(outputs["category_items"])
+        ]
+        return outputs
+
+    def convert_to_otx_format(self, datumaro_dataset: dict) -> Tuple[DatasetEntity, LabelSchemaEntity]:
+        """Convert DatumaroDataset to DatasetEntity for Acion tasks."""
+        raise NotImplementedError
+
+
+class ActionClassificationDatasetAdapter(ActionBaseDatasetAdapter):
+    """Action classification adapter inherited by ActionBaseDatasetAdapter and BaseDatasetAdapter."""
+
+    def convert_to_otx_format(self, datumaro_dataset: dict) -> Tuple[DatasetEntity, LabelSchemaEntity]:
+        """Convert DatumaroDataset to DatasetEntity for Acion Classification."""
+        label_information = self._prepare_label_information(datumaro_dataset)
+        label_entities = label_information["label_entities"]
+
+        label_schema = self._generate_default_label_schema(label_entities)
+        dataset_items = []
+        for subset, subset_data in datumaro_dataset.items():
+            for datumaro_items in subset_data:
+                for datumaro_item in datumaro_items:
+                    image = Image(file_path=datumaro_item.media.path)
```

Comment:
I think the contents of this `for statement` can also be used separately as a function.